spec/presenters/coronavirus/sub_section_json_presenter_spec.rb

```diff
@@ -245,6 +245,88 @@
     end
   end
 
+  describe "#sub_sections" do
```

Comment:
Umm, 'should' they be private is an interesting discussion - I won't go there!  Can they be?  Yes, of course - they aren't used outside of this class as you suggest.  My preference however is to avoid private methods - because: how difficult they are to test; there seems to be few genuine uses for private methods; encapsulation only really makes sense in a statically-typed language; and well-tested public methods are much easier to create and maintain.  Having said all that, I'm happy to make these methods private if there is a good reason.

Reply:
> Umm, 'should' they be private is an interesting discussion - I won't go there! 

Oh no, but you did anyway 🙈 

> Can they be? Yes, of course - they aren't used outside of this class as you suggest. My preference however is to avoid private methods - because: how difficult they are to test; there seems to be few genuine uses for private methods; encapsulation only really makes sense in a statically-typed language; and well-tested public methods are much easier to create and maintain. Having said all that, I'm happy to make these methods private if there is a good reason.

I see, your preference seems at odds with long standing GOV.UK and [community](https://rubystyle.guide/#visibility) conventions, to have a distinction between the publicly used API and the internal one, with the public one tested. So it shouldn't really be the case of a preference having an impact here, by foregoing them the code is surprising.

I'm not convinced by your reasons for the preference I'm afraid, it seems like it's going even further into the "interesting discussion" if I rebut them, basic one is that if it's hard to test it's probably doing much (harking back to [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single-responsibility_principle) - happy to have a chat if that helps though.

This class seems to have ended up in a rather odd testing place with it's lack of private / public API distinction, only some of the methods are tested, some are tested implicitly through `#output` (which I think is the lone public API method) and some things are tested twice. I hadn't actually realised how big a mess it seemed to be in until I looked now, this should serve as a good example of a class that tries to do too much and the odd testing approach concealed that.

Reply:
I totally agree with you - this class (and the tests) are messy and confused. I don't think private methods are going to help much - it needs a complete overhaul. Should we create a separate story to resolve this?

Reply:
Just looking into your ivar suggestion elsewhere, and it looks like [this](https://github.com/alphagov/collections-publisher/blob/ee9a09c16e140e3595df50a8d56596a70432a710/app/services/coronavirus/pages/sub_section_processor.rb) is another example where `output` is probably the only 'real' public method!

Reply:
> ~Just looking into your ivar suggestion elsewhere, and it looks like [this](https://github.com/alphagov/collections-publisher/blob/ee9a09c16e140e3595df50a8d56596a70432a710/app/services/coronavirus/pages/sub_section_processor.rb) is another example where `output` is probably the only 'real' public method!~ Everything goes through `.call`!

Ah `.call` seems to be a common predictable pattern. It doesn't look like this class uses `.call` though, but I may have my wires crossed about what you're saying.

Yeah, if you want to jot a quick draft story that sounds a good idea. We can then discuss at our next dev chat whether we want to do it or tackle any pain it causes in a different manner. One thing to note is that this class should have lost a bunch of it's responsibilities with the work that Jon has just merged and I think @leenagupte is talking about a follow up story to remove the the auto sub-topic paths, which we might be able to piggy back a refactor of the interface and tests onto.

Reply:
Ah, sorry if I confused. The comment was about `app/services/coronavirus/pages/sub_section_processor.rb` - which has a class method (`call`) and an `output` method.  `call` calls `new.output`. The bottom line is I think that all the other methods in this class are also candidates for being `private`.  Maybe another class that needs a bit of TLC?

Reply:
I've [put a card in](https://trello.com/c/B6v2ONY0/733-placeholder-clean-up-subsectionprocessor-and-subsectionjsonpresenter-in-collections-publisher) and included both classes.